akka-actor-typed-tests/src/test/java/jdocs/akka/typed/RouterTest.java

```diff
@@ -138,6 +191,52 @@ public DoBroadcastLog(String text) {
     // #group
   }
 
+  @Test
+  public void showGroupRoutingWithConsistentHashing() throws Exception {
+
+    TestProbe<String> probe1 = testKit.createTestProbe();
+    TestProbe<String> probe2 = testKit.createTestProbe();
+
+    Proxy proxy = new Proxy();
+
+    ActorRef<Proxy.Command> proxy1 = testKit.spawn(proxy.create(probe1.ref()));
+    ActorRef<Proxy.Command> proxy2 = testKit.spawn(proxy.create(probe2.ref()));
+
+    TestProbe<Receptionist.Registered> waiterProbe = testKit.createTestProbe();
+    // registering proxies
+
+    testKit
+        .system()
+        .receptionist()
+        .tell(Receptionist.register(proxy.registeringKey, proxy1, waiterProbe.ref()));
+    testKit
+        .system()
+        .receptionist()
+        .tell(Receptionist.register(proxy.registeringKey, proxy2, waiterProbe.ref()));
+    // wait until both registrations get Receptionist.Registered
+
+    waiterProbe.receiveSeveralMessages(2);
+    // messages sent to a router with constant hashing
+    // #constant-hashing
+    ActorRef<Proxy.Command> router =
+        testKit.spawn(
+            Routers.group(proxy.registeringKey)
+                .withConsistentHashingRouting(10, command -> proxy.mapping(command)));
+
+    router.tell(new Proxy.Message("123", "Text1"));
+    router.tell(new Proxy.Message("123", "Text2"));
+
+    router.tell(new Proxy.Message("zh3", "Text3"));
+    router.tell(new Proxy.Message("zh3", "Text4"));
+    // the hash is calculated over the Proxy.Message first parameter obtained through the
+    // Proxy.mapping function
+    // #constant-hashing
+    // Then messages with equal Message.id reach the same actor
+    // so the first message in each probe queue is equal to its second
+    probe1.expectMessage(probe1.receiveMessage());
+    probe2.expectMessage(probe2.receiveMessage());
```

Comment:
I'm confused. How is that even working? Are these tests being executed? 

The second message is not equal to the first one. It has the same id, but the content is "Text2".

Reply:
btw, `Message` is not defining equals either.

Reply:
Ok, sorry. I missed one important thing. We only send the `id` to the probe and that's what is then verified. 

I run the tests and they are working. Sorry, I'm confusing the things here. 🤦

Reply:
np at all! thanks for the effort of reviewing it @octonato